action_server/src/sema4ai/action_server/_actions_run.py

```diff
@@ -222,34 +230,49 @@ def _run_action_in_thread(
                     reuse_process,
                 )
 
-                if returncode == 0:
+                try:
+                    run_result_str: str = result_json.read_text("utf-8", "replace")
+                except Exception:
+                    raise RuntimeError(
+                        "It was not possible to collect the contents of the "
+                        "result (json not created)."
+                    )
+
+                try:
+                    result_contents = json.loads(run_result_str)
+                except Exception:
+                    raise RuntimeError(
+                        f"Error loading the contents of {run_result_str} as json."
+                    )
+
+                ret = result_contents.get("result")
+                result_str: str
+                if ret is None:
+                    result_str = "None"
+                else:
+                    result_str = json.dumps(ret, indent=4)
```

Comment:
```suggestion
                result_str = json.dumps(result_contents.get("result"), indent=4)
```
isn't this simpler and more clean?

Example:
```python
>>> json.dumps({}.get("x"))
'null'
```

Reply:
I was thinking about that too (it breaks tests, which is why I haven't done it, but I may end up doing that, just need to think a bit more about it).